README.md

```diff
@@ -62,6 +64,35 @@ should not require general ternary matches on the destination address; the
 constraint documents this intend and let's us catch accidental ternary matches
 installed by the control plane at runtime.
 
+## API
```

Comment:
This is awesome, thanks for documenting this. I still feel like we should probably reoganize the files a bit and have a top-level p4constraints.h that has the main API, so people can only include and use that. WDYT? That can be a separate PR, but we should at least track it in an issue.

Reply:
(I find the includes right now to be super random)

Reply:
Oh, I see. Yes, happy to do that. Not sure what the C++ conventions for this kind of stuff are.

Reply:
I'm also not sure, but the imports are super random, these are definitly not the files I would go look for. We should see some other libraries maybe for inspiration. But generally I think just having a single .h file that is clearly the main API file, and everything else be a bit more "hidden" seems like a good start?

Reply:
Yes, I agree with the sentiment.

Reply:
I looked into this briefly, and it seems that C++ has no standard mechanism for doing this kind of stuff. In particular, C++ does not support function aliases.

I would love to be wrong about this, but my general feeling is that C++ wasn't really build with modularity in mind (though hopefully modules will make it into C++20).

Reply:
Since I don't see an immediate fix, let's leave this for another time; we already have issue #18 to track it.